link/perf_event.go

```diff
@@ -52,11 +52,19 @@ const (
 	perfAllThreads = -1
 )
 
+type perfEventType uint32
+
+const (
```

Comment:
I prefer shorter identifiers:

```go
const (
	tracepointEvent perfEventType = iota
	kprobeEvent
	kretprobeEvent
)
```

Tracepoint can be the zero value, the event type should always be specified in a perfEvent constructor.

This enum order allows uprobe to be neatly appended later.